### PR TITLE
No redirect for /spenden 

### PR DESCRIPTION
**Pull request by kulla**
Resolves the bug that `/spenden` is redirecting to `/21565/spenden` because there is still an old page. Due to the new alias system a redirect would be triggered. It also implements the logic that `/spenden` is treated like `/search` (where the cookie determines the backend).

### CODE REVIEW

File: __tests__/__utils__/index.ts [hunk 1]
```diff
@@ -23,3 +23,12 @@
...
 export * from './services'
+
+export enum Backend {
+  Frontend = 'frontend',
+  Legacy = 'legacy',
+}
+
+export function setupProbabilityFor(backend: Backend) {
+  global.FRONTEND_PROBABILITY = backend === Backend.Frontend ? '1' : '0'
+}
```
Comment (kulla) [inline __tests__/__utils__/index.ts:34]: I noticed that when a test suite like `maintenance.ts` imports from `frontend-proxy.ts` the tests from `frontend-proxy.ts` are executed twice. I'll do a follow up PR which cleans up the utility functions.

File: __tests__/frontend-proxy.ts [hunk 2]
```diff
@@ -120,6 +117,15 @@ describe('handleRequest()', () => {
...
     })
 
+    test('prepends language prefix for special path /spenden', async () => {
+      setupProbabilityFor(Backend.Frontend)
+
+      await expectResponseFrom({
+        backend: 'https://frontend.serlo.org/en/spenden',
+        request: 'https://en.serlo.org/spenden',
+      })
+    })
```
Comment (kulla) [inline __tests__/frontend-proxy.ts:127]: Works now: https://frontend.serlo.org/de/spenden

**Review by CarolinJaser — CHANGES_REQUESTED** [review]
Thanks Kulla, I only have one small comment.

File: src/frontend-proxy.ts [hunk 3]
```diff
@@ -21,6 +21,8 @@
  */
 import { Url, getCookieValue, isInstance, Instance, getPathInfo } from './utils'
 
+export const specialPaths = ['/', '/search', '/spenden']
```
[the remark this answers is not in the document]
Comment (kulla) [reply src/frontend-proxy.ts:24]: Fixed in https://github.com/serlo/serlo.org-cloudflare-worker/pull/98

**Comment by kulla** [comment]
@kulla Ready for merge, have implemented your code suggestion in a new PR. Feel free to merge :smile:

File: src/frontend-proxy.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (CarolinJaser) [reply src/frontend-proxy.ts:24]: Ok, great, thanks!

**Review by CarolinJaser — APPROVED** [review]

**Pull request merged**